### PR TITLE
fix: [QueryBuilder] select() with RawSql may cause TypeError

### LINKED ISSUE #9007: Bug: [QueryBuilder] select() TypeError: trim(): Argument #1 ($string) must be of type string, CodeIgniter\Database\RawSql given
### PHP Version

8.1

### CodeIgniter4 Version

4.5.3

### CodeIgniter4 Installation Method

Composer (as dependency to an existing project)

### Which operating systems have you tested for this bug?

Windows, Linux

### Which server did you use?

apache

### Database

MySQL (ver. 8.3.0)

### What happened?

I encountered a TypeError when running a test that involves selecting columns using RawSql in combination with string column names. The error message indicates that the `trim()` function is receiving a `CodeIgniter\Database\RawSql` object instead of a string.

```
TypeError: trim(): Argument #1 ($string) must be of type string, CodeIgniter\Database\RawSql given

/vendor/codeigniter4/framework/system/Database/BaseBuilder.php:415
/vendor/codeigniter4/framework/system/Model.php:930
```

### Steps to Reproduce

```php

$query = model(Model::class)
->select([
    new RawSql('col1'),
    'col2'
])->builder()->getCompiledSelect();

print_r($query);
```

### Expected Output

```sql
SELECT `col1`, `col2`
FROM `table`
```

### Anything else?

If only `RawSql` objects are passed to the select method, it works fine. However, when combining `RawSql` objects with strings, the error occurs.

I temporarily resolved the issue by chaining the select method calls separately for RawSql and string columns: 
```php

$query = model(Model::class)
->select(new RawSql('col1'))
->select('col2')
->builder()->getCompiledSelect();

print_r($query);
```

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes  #9007

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: tests/system/Database/Builder/SelectTest.php [hunk 1]
```diff
@@ -67,6 +67,21 @@ public function testSelectAcceptsArray(): void
         $this->assertSame($expected, str_replace("\n", ' ', $builder->getCompiledSelect()));
     }
 
+    public function testSelectAcceptsArrayWithRawSql(): void
```
Comment (maniaba) [inline tests/system/Database/Builder/SelectTest.php:70]: Try to reverse the order and you won't get a good query, the "employee_id" column will be missing:

```php
        $builder->select([
            new RawSql("IF(salary > 5000, 'High', 'Low') AS salary_level"),
            'employee_id',
        ]);
```

```sql
-'SELECT IF(salary > 5000, 'High', 'Low') AS salary_level, "employee_id" FROM "employees"'
+'SELECT IF(salary > 5000, 'High', 'Low') AS salary_level FROM "employees"'
```

**Review by maniaba — CHANGES_REQUESTED** [review]

File: tests/system/Database/Builder/SelectTest.php (hunk 1 above)
Comment (kenjis) [reply tests/system/Database/Builder/SelectTest.php:70]: Fixed.

Comment (maniaba) [reply tests/system/Database/Builder/SelectTest.php:70]: I think it's not work again, try:

```php

$builder->select([
      new RawSql("IF(salary > 5000, 'High', 'Low') AS salary_level"),
      new RawSql("IF(test > 1, 'YES', 'NO') AS yes_no"),
      'employee_id',
]);
```
```sql
-'SELECT IF(salary > 5000, 'High', 'Low') AS salary_level, IF(test > 1, 'YES', 'NO') AS yes_no, "employee_id" FROM "employees"'
+'SELECT IF(salary > 5000, 'High', 'Low') AS salary_level, IF(salary > 5000, 'High', 'Low') AS salary_level, "employee_id" FROM "employees"'
```

File: system/Database/BaseBuilder.php [hunk 2]
```diff
@@ -3060,15 +3060,17 @@ protected function compileSelect($selectOverride = false): string
...
-                $sql .= (string) $this->QBSelect[0];
             } else {
                 // Cycle through the "select" portion of the query and prep each column name.
                 // The reason we protect identifiers here rather than in the select() function
                 // is because until the user calls the from() function we don't know if there are aliases
                 foreach ($this->QBSelect as $key => $val) {
-                    $protect              = $this->QBNoEscape[$key] ?? null;
-                    $this->QBSelect[$key] = $this->db->protectIdentifiers($val, false, $protect);
+                    if ($val instanceof RawSql) {
+                        $this->QBSelect[$key] = (string) $this->QBSelect[0];
```
Comment (maniaba) [inline system/Database/BaseBuilder.php:3069]:
```php
$this->QBSelect[$key] = (string) $this->QBSelect[0];
```
Maybe this is how we should do it?
```php
$this->QBSelect[$key] = (string) $val;
```

File: tests/system/Database/Builder/SelectTest.php (hunk 1 above)
Comment (kenjis) [reply tests/system/Database/Builder/SelectTest.php]: Fixed.

File: system/Database/BaseBuilder.php (hunk 2 above)
Comment (kenjis) [reply system/Database/BaseBuilder.php]: Yes. Fixed.

**Review by maniaba — APPROVED** [review]

**Review by datamweb — APPROVED** [review]

**Review by michalsn — APPROVED** [review]

**Pull request merged**